### ISSUE TITLE
Clear cache, cookies, history on exit option

### ISSUE THREAD

**Issue by SergeyZhukovsky** [brave/browser-android-tabs#93]

**Comment by DerFlatulator** [comment]
Is there a way to just close all tabs on exit?

**Comment by SergeyZhukovsky** [comment]
@azz no there is no way there is an issue for that https://github.com/brave/browser-android-tabs/issues/80

**Comment by SergeyZhukovsky** [comment]
We have probably implement that feature on app start, because Android OS could kill your app in any moment.

**Comment by SergeyZhukovsky** [comment]
We should have a whitelist to keep all of that for certain websites.

**Comment by srirambv** [comment]
+1 from community https://community.brave.com/t/suggestion-for-automatic-history-clear-function/5016/2

**Comment by eljuno** [comment]
+1 from community https://community.brave.com/t/feature-request-android-cleary-browsing-data-on-exit/8718?u=eljuno

>it would be great if there was a setting to automatically have the browsing data automatically deleted,

**Comment by neroux** [comment]
@SergeyZhukovsky Any update on that feature?

**Comment by SergeyZhukovsky** [comment]
@neroux we haven't started it yet. We are working on video problems and sync features currently and will look on that after.

**Comment by neroux** [comment]
Thanks for the quick response @SergeyZhukovsky. Is it still under consideration or will it be implemented? If the latter would you have any (realistic) timeframe? A month, three months, half a year, a year, ....? Thanks

**Comment by neroux** [comment]
@SergeyZhukovsky, possibly any feedback regarding

> Is it still under consideration or will it be implemented? If the latter, would you have any (realistic) timeframe? A month, three months, half a year, a year, ....?

Thanks a lot

**Comment by SergeyZhukovsky** [comment]
@neroux it is hard to say, the task isn't prioritized yet, I hope it will be avail within two months.

**Comment by neroux** [comment]
@SergeyZhukovsky No worries :), didnt "expect" a precise timeframe, a rough estimation like the ~2 months is more than enough. Just wanted to know whether it is going to happen at all. Thanks a lot.

**Comment by LaurenWags** [comment]
+1 from community: https://community.brave.com/t/automatically-clear-cached-data-upon-exit/15141

**Comment by ychenier** [comment]
Hi all, @BraveSampson on Twitter sent me here. Would definitely like to be able to clear all browsing data on Android like in Firefox. Thanks!

**Comment by neroux** [comment]
👍 @ychenier

I realise, the two months mark has only just passed, so my question might come a bit early but still curious of course .... do you think that feature will be available soon (+/- 30 days) or is it still in the making and will take some more time @SergeyZhukovsky? Thanks a lot!

**Comment by Netronic** [comment]
Hello! Brave dev send me here from playstore. It would be nice to clear all browsing data when the browser close! I waiting for this a long time...

**Comment by rix1337** [comment]
I look to only disable history (not cookies) like you can on Firefox.

Should be fairly easy. Look at the Chrome desktop addon No History
https://chrome.google.com/webstore/detail/no-history/ljamgkbcojbnmcaonjokopmcblmmpfch

**Comment by ShadronSoul** [comment]
Any progress so far? This feauture would be so useful. Sad it's not there already.

**Comment by motorious1** [comment]
also VERY interested in this feature on the mobile app...hope it comes soon.

**Comment by rix1337** [comment]
How is progress on the keep history toggle? This should not be hard.

**Comment by eljuno** [comment]
+1 from community for history https://community.brave.com/t/recent-sites-on-homepage/23048?u=eljuno

**Comment by kjozwiak** [comment]
+1 from a user on the brave subreddit thttps://www.reddit.com/r/brave_browser/comments/8unged/clear_history_on_exit_brave_browser_android/

**Comment by eljuno** [comment]
+1 from community https://community.brave.com/t/first-party-cookies-and-clearing-data/29170?u=eljuno

**Comment by Brave-Matt** [comment]
+1 from Zendesk user: 

> Hi, i really like the browser.. It needs one more thing.. When you exit it needs to automatically delete cookies and visited sites..
> Others do this...
> Keep up the good work...

**Comment by neroux** [comment]
@SergeyZhukovsky, is there possibly any update on that issue? Is it still under consideration?

**Comment by neroux** [comment]
@SergeyZhukovsky How are the chances it will really show up in 1.0.61? Thanks.

**Comment by SergeyZhukovsky** [comment]
We are working on Android ledger integration and sync features. All our dev resources are there currently. It's hard to say when we will work on that, probably in couple months.

**Comment by kjozwiak** [comment]
> 1.0.61 seems to be scheduled for the end of this month. As you said "couple of months" I guess it wont be in 1.0.61 then, right?

Yup, unfortunately this feature isn't going to be added/implemented in `1.0.61` due to the dev's concentrating on Sync and adding Payments/Rewards on mobile.

**Comment by neroux** [comment]
Thank you for the confirmation, @kjozwiak

**Comment by neroux** [comment]
@SergeyZhukovsky @kjozwiak Any update maybe?

Currently it targets 1.0.62 for mid-December. Is that stilll realistic?

I'd love to switch to Brave but this feature missing is a deal-breaker I am afraid.

Thanks a lot!

**Comment by Brave-Matt** [comment]
+1 from Zendesk: ![image](https://user-images.githubusercontent.com/3299098/49042396-ee3cc980-f17c-11e8-8b17-d247ae9a976c.png)

**Comment by JohnGLFour** [comment]
I want this feature because of Brave on Linux has one. # #

**Comment by ghosted17** [comment]
Really want this for android or at least stop putting recent sites to start page. Privacy first

**Comment by quafeg** [comment]
Would really appreciate this feature.  Thank you devs for all your hard work and for making brave so great!

**Comment by caldetas** [comment]
+1

**Comment by NumDeP** [comment]
@SergeyZhukovsky silly question but considering this feature is coherently being worked for b-b, couldn't any part of the code also be utilised for this mobile chromium or be worked on simultaneously by the devs?

**Comment by SergeyZhukovsky** [comment]
Not sure I understand what is b-b. The feature is not hard and most likely it has to be implemented on Java Android related part. We will get back to it soon when finish with current critical features.

**Comment by NumDeP** [comment]
Sorry. I figured internally the team calls brave-browser - b-b. 

It's probably good you left it this late to implement anyhow, as some of the information on the critical features could also be in the checklist for erasing like in brave-browser.

**Comment by JoeBloggs2345** [comment]
Hello please can this be implemented? It will hopefully take a slight amount of your time but benefit many people.  
For example you could poach a lot of users from Firefox Focus is this is implemented.
Cheers.

**Comment by sviluppoSoftware** [comment]
please add this feature for Android !!!

**Comment by FrederickDavies** [comment]
"Clear cache, cookies, history on exit option" in the Android version.
+1

**Comment by zcxvxxcvcv** [comment]
Hi @SergeyZhukovsky  it seems this feature has been implemented on the desktop browser. Please bring this to Android as well bro. I appreciate there will need to be some tweaking since Android apps can be killed at any time but I'm sure it will be straightforward for you! Thanks :)

**Comment by laradevitt** [comment]
@zcxvxxcvcv - In the new desktop version there is a setting for "Keep local data only until you quit your browser" under Content Settings > Cookies - which apparently doesn't affect localstorage according to [this issue](https://github.com/brave/brave-browser/issues/1127) - but I don't see any options for cache or history. Am I missing something?

**Comment by zcxvxxcvcv** [comment]
@laradevitt Last I checked it was in the nightly channel so I presume it will eventually come to the stable release soon enough. You would find it in the clear browser data option option alongside the full list of items which chrome can usually clear.

**Comment by FrederickDavies** [comment]
Dear All,

	OK, my desktop's Brave got updated, but how do I check it is working or not? Where are the cookies and other caches saved in Brave? Essentially, what is the Brave equivalent of IE's AppData\Roaming\Microsoft\Windows\Cookies?

	Yours,

		FD

**Comment by fshuk1** [comment]
For Android PLEASE. I use gogoDuck for this very reason but I would like to use BRAVE. Is there a time frame for it or I should just not hold my breath?

**Comment by fshuk1** [comment]
ops DuckDuckGo

**Comment by rogueTriple** [comment]
Any updates? This issue has been open since Dec '16.

**Comment by losteh** [comment]
This is a must have if Privacy is at forefront for Brave.

Appreciate the focus is on business development to monetize but this needs to be expedited

Even adding to flags or other workaround would be better than 2 years of NO PROGRESS. Not a pop at hardworking developers, but a strong nudge at the senior management team

**Comment by Brave-Matt** [comment]
+1 from Reddit:
https://www.reddit.com/r/brave_browser/comments/c5prfj/clear_cache_and_cookies_upon_closing/

**Comment by jwsp1** [comment]
Any update on this issue? :)

**Comment by rogueTriple** [comment]
I’ve stopped using brave, you started out strong but are pretty garbage at this point.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by seocrow** [comment]
Please can this be implemented on Brave for Android. This is a security risk and would save a lot of time manually clearing.

* Automatically delete browser cache, URL history on shutdown

**Comment by Brave-Matt** [comment]
+1 from Reddit:
https://www.reddit.com/r/brave_browser/comments/dl156m/clear_cookies_on_exit_of_browser_on_android/

**Comment by deeppandya** [comment]
@SergeyZhukovsky @anthonypkeane @jamesmudgett i think we should have a switch in settings to provide control to the user.

**Comment by srirambv** [comment]
@deeppandya  could we do what we have on desktop and have a third tab which sets individual options to clear on exit?

**Comment by anthonypkeane** [comment]
@jamesmudgett to come up with a platform design here first before this is implemented as per https://bravesoftware.slack.com/archives/C0816QH3R/p1573803137402600
